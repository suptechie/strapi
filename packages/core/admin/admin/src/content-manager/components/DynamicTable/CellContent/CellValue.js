import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import toString from 'lodash/toString';

const CellValue = ({ type, value }) => {
  const { formatDate, formatTime, formatNumber } = useIntl();
  let formattedValue = value;

  if (type === 'date') {
    formattedValue = formatDate(value, { dateStyle: 'full' });
  }

  if (type === 'datetime') {
    formattedValue = formatDate(value, { dateStyle: 'full', timeStyle: 'short' });
  }

  if (type === 'time') {
    const [hour, minute, second] = value.split(':');
    const date = new Date();
    date.setHours(hour);
    date.setMinutes(minute);
    date.setSeconds(second);

    formattedValue = formatTime(date, {
      numeric: 'auto',
      style: 'short',
    });
  }

  if (['float', 'decimal'].includes(type)) {
    const numberOfDecimals = getNumberOfDecimals(value);

    formattedValue = formatNumber(value, {
      minimumFractionDigits: numberOfDecimals,
      maximumFractionDigits: numberOfDecimals,
    });
  }

  if (['integer', 'biginteger'].includes(type)) {
    formattedValue = formatNumber(value, { maximumFractionDigits: 0 });
  }

  return toString(formattedValue);
};

const getNumberOfDecimals = value => {
  if (value % 1 !== 0) {
    // value has decimals
    return value.toString().split('.')[1].length;
  }

  return 0;
};

CellValue.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.any.isRequired,
};

export default CellValue;
