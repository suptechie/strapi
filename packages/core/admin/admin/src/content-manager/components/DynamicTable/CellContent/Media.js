import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Avatar } from '@strapi/parts/Avatar';
import { Text } from '@strapi/parts/Text';
import { Row } from '@strapi/parts/Row';
import { getFileExtension, prefixFileUrlWithBackendUrl } from '@strapi/helper-plugin';

// TODO: this is very temporary until we get a design
const FileWrapper = styled(Row)`
  position: relative;
  border-radius: 50%;
  width: 26px;
  height: 26px;
  background: ${({ theme }) => theme.colors.neutral100};
`;

const Media = ({ url, mime, alternativeText, name, ext, formats }) => {
  const fileURL = prefixFileUrlWithBackendUrl(url);

  if (mime.includes('image')) {
    const thumbnail = formats?.thumbnail?.url || null;
    const mediaURL = prefixFileUrlWithBackendUrl(thumbnail) || fileURL;

    return <Avatar src={mediaURL} alt={alternativeText || name} preview />;
  }

  const fileExtension = getFileExtension(ext);

  return (
    <FileWrapper>
      <Text textColor="neutral800">{fileExtension}</Text>
    </FileWrapper>
  );
};

Media.defaultProps = {
  alternativeText: null,
  formats: null,
};

Media.propTypes = {
  alternativeText: PropTypes.string,
  ext: PropTypes.string.isRequired,
  formats: PropTypes.object,
  mime: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default Media;
