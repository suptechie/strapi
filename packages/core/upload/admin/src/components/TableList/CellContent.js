import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { getFileExtension } from '@strapi/helper-plugin';
import { Typography } from '@strapi/design-system/Typography';

import { PreviewCell } from './PreviewCell';
import { formatBytes } from '../../utils';

export const CellContent = ({
  alternativeText,
  content,
  cellType,
  elementType,
  mime,
  fileExtension,
  thumbnailURL,
  url,
}) => {
  const { formatDate } = useIntl();
  switch (cellType) {
    case 'image':
      return (
        <PreviewCell
          alternativeText={alternativeText}
          fileExtension={fileExtension}
          mime={mime}
          type={elementType}
          thumbnailURL={thumbnailURL}
          url={url}
        />
      );
    case 'date':
      return <Typography>{formatDate(content)}</Typography>;

    case 'size':
      if (elementType === 'folder') return <Typography>-</Typography>;

      return <Typography>{formatBytes(content)}</Typography>;

    case 'ext':
      if (elementType === 'folder') return <Typography>-</Typography>;

      return <Typography>{getFileExtension(content).toUpperCase()}</Typography>;

    case 'text':
      return <Typography>{content}</Typography>;

    default:
      return <Typography>-</Typography>;
  }
};

CellContent.defaultProps = {
  alternativeText: null,
  content: '',
  fileExtension: '',
  mime: '',
  thumbnailURL: null,
  url: null,
};

CellContent.propTypes = {
  alternativeText: PropTypes.string,
  content: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  fileExtension: PropTypes.string,
  mime: PropTypes.string,
  thumbnailURL: PropTypes.string,
  cellType: PropTypes.string.isRequired,
  elementType: PropTypes.string.isRequired,
  url: PropTypes.string,
};
