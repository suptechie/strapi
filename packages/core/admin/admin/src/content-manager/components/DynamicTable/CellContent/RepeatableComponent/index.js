import React, { useState, useRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { useIntl } from 'react-intl';
import { Flex } from '@strapi/design-system/Flex';
import { Box } from '@strapi/design-system/Box';
import { Tooltip } from '@strapi/design-system/Tooltip';
import { Typography } from '@strapi/design-system/Typography';
import { Popover } from '@strapi/design-system/Popover';
import { FocusTrap } from '@strapi/design-system/FocusTrap';
import { SortIcon, stopPropagation } from '@strapi/helper-plugin';
import { getTrad } from '../../../../utils';

import CellValue from '../CellValue';

const Button = styled.button`
  svg {
    > g,
    path {
      fill: ${({ theme }) => theme.colors.neutral500};
    }
  }
  &:hover {
    svg {
      > g,
      path {
        fill: ${({ theme }) => theme.colors.neutral600};
      }
    }
  }
  &:active {
    svg {
      > g,
      path {
        fill: ${({ theme }) => theme.colors.neutral400};
      }
    }
  }
`;

const ActionWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${32 / 16}rem;
  width: ${32 / 16}rem;
  svg {
    height: ${4 / 16}rem;
  }
`;

const RepeatableComponentCell = ({ value, metadatas }) => {
  const [visible, setVisible] = useState(false);
  const { formatMessage } = useIntl();
  const buttonRef = useRef();
  const {
    mainField: { type: mainFieldType, name: mainFieldName },
  } = metadatas;
  const subItems = [...value.slice(1)];

  const handleTogglePopover = () => setVisible(prev => !prev);

  return (
    <Flex {...stopPropagation}>
      <Tooltip
        label={formatMessage({
          id: getTrad('DynamicTable.component.repeatable.toggle'),
          defaultMessage: 'Toggle repeatable values',
        })}
      >
        <Button type="button" onClick={handleTogglePopover} ref={buttonRef}>
          <Flex>
            <Typography
              style={{ maxWidth: '252px', cursor: 'pointer' }}
              textColor="neutral800"
              ellipsis
            >
              <CellValue type={mainFieldType} value={value[0][mainFieldName] || '[TBD]'} />
            </Typography>
            <ActionWrapper>
              <SortIcon />

              {visible && subItems.length > 0 && (
                <Popover source={buttonRef} spacing={16} centered>
                  <FocusTrap onEscape={handleTogglePopover}>
                    <ul>
                      {subItems.map(item => (
                        <Box key={item.id} tabIndex={0} padding={3} as="li">
                          <Typography>
                            <CellValue type={mainFieldType} value={item[mainFieldName]} />
                          </Typography>
                        </Box>
                      ))}
                    </ul>
                  </FocusTrap>
                </Popover>
              )}
            </ActionWrapper>
          </Flex>
        </Button>
      </Tooltip>
    </Flex>
  );
};

RepeatableComponentCell.propTypes = {
  metadatas: PropTypes.shape({
    mainField: PropTypes.shape({
      name: PropTypes.string,
      type: PropTypes.string,
      value: PropTypes.string,
    }),
  }).isRequired,
  value: PropTypes.array.isRequired,
};

export default RepeatableComponentCell;
