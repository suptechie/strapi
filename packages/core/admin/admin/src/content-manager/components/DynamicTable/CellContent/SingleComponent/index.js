import React from 'react';
import PropTypes from 'prop-types';
import { Tooltip } from '@strapi/design-system/Tooltip';
import { Typography } from '@strapi/design-system/Typography';

import CellValue from '../CellValue';

const SingleComponentCell = ({ value, metadatas }) => {
  const { mainField } = metadatas;
  const content = value[mainField.name];

  return (
    <Tooltip label={content}>
      <Typography textColor="neutral800" ellipsis>
        <CellValue type={mainField.type} value={content} />
      </Typography>
    </Tooltip>
  );
};

SingleComponentCell.propTypes = {
  metadatas: PropTypes.shape({
    mainField: PropTypes.shape({
      name: PropTypes.string,
      type: PropTypes.string,
      value: PropTypes.string,
    }),
  }).isRequired,
  value: PropTypes.object.isRequired,
};

export default SingleComponentCell;
