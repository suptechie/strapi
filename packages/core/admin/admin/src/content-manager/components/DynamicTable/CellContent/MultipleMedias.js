import React from 'react';
import PropTypes from 'prop-types';
import { AvatarGroup } from '@strapi/parts/Avatar';
import Media from './Media';

const MultipleMedia = ({ value }) => {
  return (
    <AvatarGroup>
      {value.map((file, index) => {
        const key = `${file.id}${index}`;

        return <Media key={key} {...file} />;
      })}
    </AvatarGroup>
  );
};

MultipleMedia.propTypes = {
  value: PropTypes.arrayOf(
    PropTypes.shape({
      alternativeText: PropTypes.string,
      ext: PropTypes.string.isRequired,
      formats: PropTypes.object,
      mime: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default MultipleMedia;
