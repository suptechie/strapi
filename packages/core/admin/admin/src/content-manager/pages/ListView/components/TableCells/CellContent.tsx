import { Tooltip, Typography } from '@strapi/design-system';
import isEmpty from 'lodash/isEmpty';
import styled from 'styled-components';

import { useDoc } from '../../../../hooks/useDocument';

import { CellValue } from './CellValue';
import { SingleComponent, RepeatableComponent } from './Components';
import { MediaSingle, MediaMultiple } from './Media';
import { RelationMultiple, RelationSingle } from './Relations';

import type { ListFieldLayout } from '../../../../hooks/useDocumentLayout';
import type { Attribute, Entity } from '@strapi/types';

interface CellContentProps extends Omit<ListFieldLayout, 'cellFormatter'> {
  content: Attribute.GetValue<Attribute.Any>;
  rowId: Entity.ID;
}

const CellContent = ({ content, mainField, attribute, rowId, name }: CellContentProps) => {
  const { components } = useDoc();

  if (!hasContent(content, mainField, attribute)) {
    return <Typography textColor="neutral800">-</Typography>;
  }

  switch (attribute.type) {
    case 'media':
      if (!attribute.multiple) {
        return <MediaSingle {...content} />;
      }

      return <MediaMultiple content={content} />;

    case 'relation': {
      if (isSingleRelation(attribute.relation)) {
        return <RelationSingle mainField={mainField} content={content} />;
      }

      return <RelationMultiple rowId={rowId} mainField={mainField} content={content} name={name} />;
    }

    case 'component':
      if (attribute.repeatable) {
        return (
          <RepeatableComponent
            schema={components[attribute.component]}
            mainField={mainField}
            content={content}
          />
        );
      }

      return (
        <SingleComponent
          schema={components[attribute.component]}
          mainField={mainField}
          content={content}
        />
      );

    case 'string':
      return (
        <Tooltip description={content}>
          <TypographyMaxWidth ellipsis textColor="neutral800">
            <CellValue type={attribute.type} value={content} />
          </TypographyMaxWidth>
        </Tooltip>
      );

    default:
      return (
        <TypographyMaxWidth ellipsis textColor="neutral800">
          <CellValue type={attribute.type} value={content} />
        </TypographyMaxWidth>
      );
  }
};

const TypographyMaxWidth = styled(Typography)`
  max-width: 300px;
`;

const hasContent = (
  content: CellContentProps['content'],
  mainField: CellContentProps['mainField'],
  attribute: CellContentProps['attribute']
) => {
  if (attribute.type === 'component') {
    // Repeatable fields show the ID as fallback, in case the mainField
    // doesn't have any content
    if (attribute.repeatable || !mainField) {
      return content?.length > 0;
    }

    const value = content?.[mainField];

    // relations, media ... show the id as fallback
    if (mainField === 'id' && ![undefined, null].includes(value)) {
      return true;
    }

    /* The ID field reports itself as type `integer`, which makes it
       impossible to distinguish it from other number fields.

       Biginteger fields need to be treated as strings, as `isNumber`
       doesn't deal with them.
    */
    // if (
    //   isFieldTypeNumber(mainField.type) &&
    //   mainField.type !== 'biginteger' &&
    //   mainField.name !== 'id'
    // ) {
    //   return typeof value === 'number';
    // }

    return !isEmpty(value);
  }

  if (attribute.type === 'relation') {
    if (isSingleRelation(attribute.relation)) {
      return !isEmpty(content);
    }

    return content?.count > 0;
  }

  /* 
      Biginteger fields need to be treated as strings, as `isNumber`
      doesn't deal with them.
  */
  if (['integer', 'decimal', 'float', 'number'].includes(attribute.type)) {
    return typeof content === 'number';
  }

  if (attribute.type === 'boolean') {
    return content !== null;
  }

  return !isEmpty(content);
};

const isSingleRelation = (
  type: Extract<CellContentProps['attribute'], { type: 'relation' }>['relation']
) => ['oneToOne', 'manyToOne', 'oneToOneMorph'].includes(type);

export { CellContent };
export type { CellContentProps };
