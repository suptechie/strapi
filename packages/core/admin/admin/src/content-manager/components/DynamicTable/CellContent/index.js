import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Typography } from '@strapi/design-system/Typography';
import Media from './Media';
import MultipleMedias from './MultipleMedias';
import Relation from './Relation';
import RepeatableComponent from './RepeatableComponent';
import CellValue from './CellValue';
import hasContent from './utils/hasContent';

const TypographyMaxWidth = styled(Typography)`
  max-width: 300px;
`;

const CellContent = ({ content, fieldSchema, metadatas, name, queryInfos, rowId, layout }) => {
  const { type } = fieldSchema;

  if (!hasContent(type, content, metadatas, fieldSchema)) {
    return <Typography textColor="neutral800">-</Typography>;
  }

  switch (type) {
    case 'media':
      if (!fieldSchema.multiple) {
        return <Media {...content} />;
      }

      return <MultipleMedias value={content} />;

    case 'relation':
      return (
        <Relation
          fieldSchema={fieldSchema}
          queryInfos={queryInfos}
          metadatas={metadatas}
          value={content}
          name={name}
          rowId={rowId}
        />
      );

    case 'component':
      return (
        <RepeatableComponent
          value={content}
          metadatas={metadatas}
          component={layout?.components?.[fieldSchema.component]}
        />
      );

    default:
      return (
        <TypographyMaxWidth ellipsis textColor="neutral800">
          <CellValue type={type} value={content} />
        </TypographyMaxWidth>
      );
  }
};

CellContent.defaultProps = {
  content: undefined,
  queryInfos: undefined,
};

CellContent.propTypes = {
  content: PropTypes.any,
  fieldSchema: PropTypes.shape({
    component: PropTypes.string,
    multiple: PropTypes.bool,
    type: PropTypes.string.isRequired,
    repeatable: PropTypes.bool,
  }).isRequired,
  metadatas: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
  rowId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  queryInfos: PropTypes.shape({ endPoint: PropTypes.string.isRequired }),
  layout: PropTypes.shape({
    components: PropTypes.object,
  }).isRequired,
};

export default CellContent;
