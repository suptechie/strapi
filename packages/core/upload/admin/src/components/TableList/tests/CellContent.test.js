import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';

import { CellContent } from '../CellContent';

const PROPS_FIXTURE = {
  alternativeText: 'alternative alt',
  cellType: 'image',
  elementType: 'asset',
  content: 'michka-picture-url-default.jpeg',
  fileExtension: '.jpeg',
  mime: 'image/jpeg',
  thumbnailURL: 'michka-picture-url-thumbnail.jpeg',
  url: 'michka-picture-url-default.jpeg',
};

const ComponentFixture = (props) => {
  const customProps = {
    ...PROPS_FIXTURE,
    ...props,
  };

  return (
    <IntlProvider locale="en" messages={{}}>
      <ThemeProvider theme={lightTheme}>
        <CellContent {...PROPS_FIXTURE} {...customProps} />
      </ThemeProvider>
    </IntlProvider>
  );
};

const setup = (props) => render(<ComponentFixture {...props} />);

describe('TableList | CellContent', () => {
  it('should render image cell type when element type is asset and mime includes image', () => {
    const { getByRole } = setup();

    expect(getByRole('img', { name: 'alternative alt' })).toBeInTheDocument();
  });

  it('should render image cell type when element type is asset and mime does not include image', () => {
    const { getByText } = setup({ mime: 'application/pdf', fileExtension: 'pdf' });

    expect(getByText('pdf')).toBeInTheDocument();
  });

  it('should render image cell type when element type is folder', () => {
    const { container } = setup({ elementType: 'folder' });

    expect(container.querySelector('path')).toBeInTheDocument();
  });

  it('should render text cell type', () => {
    const { getByText } = setup({ cellType: 'text', content: 'some text' });

    expect(getByText('some text')).toBeInTheDocument();
  });

  it('should render extension cell type', () => {
    const { getByText } = setup({ cellType: 'ext', content: '.pdf' });

    expect(getByText('PDF')).toBeInTheDocument();
  });

  it('should render size cell type', () => {
    const { getByText } = setup({ cellType: 'size', content: '20.5435' });

    expect(getByText('21KB')).toBeInTheDocument();
  });

  it('should render date cell type', () => {
    const { getByText } = setup({ cellType: 'date', content: '2022-11-18T12:08:02.202Z' });

    expect(getByText('11/18/2022')).toBeInTheDocument();
  });
});
