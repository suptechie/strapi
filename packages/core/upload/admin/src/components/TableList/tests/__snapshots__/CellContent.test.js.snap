// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableList | CellContent should render "-" by default when no recognized cell type is passed 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

<div>
  <span
    class="c0"
  >
    -
  </span>
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render date cell type 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

<div>
  <span
    class="c0"
  >
    11/18/2022
  </span>
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render extension cell type when element type is asset 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

<div>
  <span
    class="c0"
  >
    PDF
  </span>
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render extension cell type with "-" when element type is folder 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

<div>
  <span
    class="c0"
  >
    -
  </span>
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render image cell type when element type is asset and mime does not include image 1`] = `
.c4 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  background: #eaf5ff;
  border-radius: 50%;
  width: 1.625rem;
  height: 1.625rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #0c75af;
}

.c2 span {
  font-size: 0.625rem;
}

<div>
  <div
    class="c0 c1 c2"
    height="1.625rem"
    width="1.625rem"
  >
    <span
      class="c3"
    >
      pdf
    </span>
  </div>
  <div
    class="c4"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render image cell type when element type is asset and mime includes image 1`] = `
.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c1 {
  border-radius: 50%;
  object-fit: cover;
  display: block;
  position: relative;
}

.c0 {
  position: relative;
  width: 1.625rem;
  height: 1.625rem;
}

<div>
  <span>
    <div
      class="c0"
    >
      <img
        alt="alternative alt"
        class="c1"
        height="26px"
        src="michka-picture-url-thumbnail.jpeg"
        width="26px"
      />
    </div>
  </span>
  <div
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render image cell type when element type is folder 1`] = `
.c4 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  background: #eaf5ff;
  border-radius: 50%;
  width: 1.625rem;
  height: 1.625rem;
}

.c2 {
  color: #66b7f1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 path {
  fill: #66b7f1;
}

<div>
  <div
    class="c0 c1"
    height="1.625rem"
    width="1.625rem"
  >
    <svg
      class="c2 c3"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.414 5H21a1 1 0 011 1v14a1 1 0 01-1 1H3a1 1 0 01-1-1V4a1 1 0 011-1h7.414l2 2z"
        fill="#212134"
      />
    </svg>
  </div>
  <div
    class="c4"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render size cell type when element type is asset 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

<div>
  <span
    class="c0"
  >
    21KB
  </span>
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render size cell type with "-" when element type is folder 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

<div>
  <span
    class="c0"
  >
    -
  </span>
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`TableList | CellContent should render text cell type 1`] = `
.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #32324d;
}

<div>
  <span
    class="c0"
  >
    some text
  </span>
  <div
    class="c1"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
